x-pack/plugins/task_manager/server/polling/task_poller.ts

```diff
@@ -151,7 +151,14 @@ export enum PollingErrorType {
 }
 
 function asPollingError<T>(err: Error, type: PollingErrorType, data: Option<T> = none) {
-  return asErr(new PollingError<T>(`Failed to poll for work: ${err.message}`, type, data, err));
+  return asErr(
+    new PollingError<T>(
+      `Failed to poll for work: ${err.message || err}`,
+      type,
+      data,
+      typeof err === 'string' ? new Error(err) : err
```

Comment:
Maybe we should reverse this test?  Test for instanceof Error instead?  That way we'd catch things besides strings being thrown - since we don't really know *WHAT* is being thrown ...

```suggestion
      err instanceof Error ? err : new Error(`${err}`)
```

Reply:
Makes sense to me, I've added your suggestion.